### ISSUE TITLE
Infinite loop in davix slow unit tests when using external libcurl

### ISSUE THREAD

**Issue by amadio** [cern-fts/davix#98]
The test seems to hang because of a version mismatch between strings in the test and the actual library.
I had to apply the following change for the tests to pass for me:
```diff
--- a/test/slow-unit/test-utils.hpp
+++ b/test/slow-unit/test-utils.hpp
@@ -33,7 +33,7 @@
 #define SSTR(message) static_cast<std::ostringstream&>(std::ostringstream().flush() << message).str()
 
 inline std::string getCurlUserAgent() {
-  return SSTR("User-Agent: " << Davix::RequestParams().getUserAgent() << " libcurl/7.69.0-DEV\r\n");
+  return SSTR("User-Agent: " << Davix::RequestParams().getUserAgent() << " libcurl/7.85.0\r\n");
 }
 
 inline std::string getDefaultUserAgent() {
```
I was compiling against curl-7.85.0. This string may have to be generated for the right library version for the test to always work.

**Comment by mpatrascoiu** [comment]
Hello,

Thanks for pointing this. Upon merging your previous change (https://github.com/cern-fts/davix/pull/99), definitely this comes up next. We will look into providing a fix. Ultimately, we want to have the `davix-slow-unit-tests` running in our gitlab-CI as well.

Cheers,
Mihai

**Comment by joaopblopes** [comment]
Fixed with commit [50666e84](https://github.com/cern-fts/davix/commit/50666e84a5156490294bc0ddac66781bbc2af06d).
